dictionary.rb

```diff
@@ -0,0 +1,100 @@
+require_relative 'dictionary_loader'
+require_relative 'dictionary_analyzer'
+
+class Dictionary
+  attr_reader :dictionary, :analyzer, :search_results
+  def initialize
+    print "What is the name of the dictionary you want to load?\n > "
+    dictionary_name = gets.chomp
+    check_for_quit dictionary_name
+    @dictionary = DictionaryLoader.new(dictionary_name).words
+    @analyzer = DictionaryAnalyzer.new(dictionary)
+    @search_results = search
+    do_something_with_results
+  end
+
+  private
+
+  def search
+    print "\nHow would you like to search?\n1 for an exact match\n2 for partial matches\n3 to match beginning of word\n4 to match end of word\n > "
```

Comment:
You even make the terminal work for a user's eyes. I need to up my game. I just let it spill over and go all crazy like.